spec/services/google/search_service_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'support/http_client_helpers'
+
+RSpec.describe Google::SearchService, type: :service do
+  describe '#search!' do
+    context 'given a successful request' do
+      it 'returns the result page html when the request succeeds' do
+        html = '<html><title>Search results</title></html>'
+        HttpClientHelpers.stub_http_adapter do |adapter, connection|
+          allow(described_class).to receive(:http_client).and_return(connection)
+          adapter.get('/search') { [200, { 'Content-Type': 'text/html' }, html] }
+        end
+
+        expect(described_class.search!('hello world')).to eq(html)
+      end
+    end
+
+    context 'given a failed request' do
+      it 'raise GoogleScraperRuby::Errors::SearchServiceError if the request returns a 4xx status code' do
```

Comment:
`it raises`

Reply:
Fixed in 6abe0de029aea0819556cb8a42fc90354de264fa